### DISCUSSION TITLE
Why compiling debian package is slower than direclty compiling the linux kernel..?

### DISCUSSION THREAD

**Discussion by Ruach** [Joshua-Riek/ubuntu-rockchip#846] [Q&A]
Hi, I have one question about debianl build. When I compile the linux kernel with the provided script, it seems that entire cores not 100% utilized.Compared to it, when I just do below, it fully utilize 80 cores all times.

> make ARCH=arm64 -j80 

I checked the rules files and it seems that it utilize CONCURRENCY_LEVEL, so I give CONCURRENCY_LEVEL=80 with fakeroot command in the script. 

Based on the log file, it seems that -j80 has been passed, but when I do htop, it doesn't really utilize all cores for compilation 
> make ARCH=arm64 CROSS_COMPILE=aarch64-linux-gnu- HOSTCC=aarch64-linux-gnu-gcc CC=aarch64-linux-gnu-gcc KERNELVERSION=6.1.0-1014-rockchip CONFIG_DEBUG_SECTION_MISMATCH=y KBUILD_BUILD_VERSION="14" LOCALVERSION= localver-extra= CFLAGS_MODULE="-DPKG_ABI=1014" PYTHON=python3 O=/home/XX/orangepi/ubuntu-rockchip/build/linux-rockchip/debian/build/build-rockchip -j80 syncconfig prepare scripts

> make ARCH=arm64 CROSS_COMPILE=aarch64-linux-gnu- HOSTCC=aarch64-linux-gnu-gcc CC=aarch64-linux-gnu-gcc KERNELVERSION=6.1.0-1014-rockchip CONFIG_DEBUG_SECTION_MISMATCH=y KBUILD_BUILD_VERSION="14" LOCALVERSION= localver-extra= CFLAGS_MODULE="-DPKG_ABI=1014" PYTHON=python3 O=/home/XX/orangepi/ubuntu-rockchip/build/linux-rockchip/debian/build/build-rockchip -j80 Image modules dtbs

**Comment by Joshua-Riek** [comment]
I only have a system with the max of 16 cores, so I have no clue why it will not use all the cpu cores on an 80-core system. But for me, it uses all the available cpu cores.

**Reply by Ruach** [reply]
@Joshua-Riek Hmm, that's strange. By the way, whenever I compile the kernel, it appears that the entire build directory gets deleted, and the kernel is recompiled from scratch, even if I've only modified a single line in the driver. As I'm not very familiar with the Debian packaging system, is there a way to configure it to only compile the modified parts and then generate the Debian packages?
